packages/style-engine/class-wp-style-engine-css-rules.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * WP_Style_Engine_CSS_Rules
+ *
+ * Holds, sanitizes and prints CSS rules.
+ *
+ * @package Gutenberg
+ */
+
+if ( class_exists( 'WP_Style_Engine_CSS_Rules' ) ) {
+	return;
+}
+
+/**
+ * Holds, sanitizes, processes and prints CSS styles rules for the style engine.
+ *
+ * @access private
+ */
+class WP_Style_Engine_CSS_Rules {
```

Comment:
Naming is tricky with these things — just checking that I'm reading this correctly, but the scope for this object is that it deals with a flat array of key/value pairs of CSS rules, that will (separately) be attached to a selector in the style engine? Since the methods of this class are about declarations, I was wondering if it's worth renaming the class so that it's clearer it deal with declarations, and then we might have a separate store/registry like in your inline styles handler PR (https://github.com/WordPress/gutenberg/pull/41896)?

Reply:
@andrewserong 

> just checking that I'm reading this correctly, but the scope for this object is that it deals with a flat array of key/value pairs of CSS rules, that will (separately) be attached to a selector in the style engine?

That is correct.

I'm open to suggestions! Like you said, naming things is hard... 💯 
What about `WP_Style_Engine_CSS_Rules_Declarations` ? Or is that too verbose?

Reply:
I don't feel too strongly about it, but I wonder if the slightly shorter `WP_Style_Engine_CSS_Declarations` would work? At least in [MDN's description](https://developer.mozilla.org/en-US/docs/Web/CSS/Syntax), it sounds like a rule/ruleset includes the selector?

Reply:
Makes sense... Done 👍

Reply:
It looks like it's compiling and returning CSS declarations, so possibly `WP_Style_Engine_CSS_Declarations` to discern between declarations and CSS rules, which express `$selector { $css_declaration_block }`.

Since there are so many words flying about the place, we recently added a glossary of terms (basically following MDN usage).

https://github.com/WordPress/gutenberg/tree/trunk/packages/style-engine#glossary

I also don't know if it's suitable _enough_. It's hard to see if we'll ever need to extend such a class, and as smarter people than I have said: _"Once it's in, it's in"_  😄

Reply:
I'm glad we're all in agreement! `WP_Style_Engine_CSS_Declarations` makes perfect sense, and is inline with the glossary you posted above as well 🎉